### PR TITLE
Firestore: Encode document id on URI

### PR DESCRIPTION
**Pull request by edupsousa**
Small fix to show firestore documents that contains # and other special characters on the id.

### CODE REVIEW

**Comment by edupsousa** [comment]
@googlebot I signed it!

File: src/components/Firestore/Collection.tsx [hunk 1]
```diff
@@ -146,7 +146,8 @@ export const Collection: React.FC<Props> = ({ collection }) => {
       <Route
         path={`${url}/:id`}
         render={({ match }: any) => {
-          const docRef = collection.doc(match.params.id);
+          const docId = decodeURIComponent(match.params.id);
```
Comment (yuchenshi) [inline src/components/Firestore/Collection.tsx:149]: I think react router should already decode this for us, no? Please double check to make sure we're not double decoding.

File: src/components/Firestore/Collection.tsx [hunk 2]
```diff
@@ -135,7 +135,7 @@ export const Collection: React.FC<Props> = ({ collection }) => {
               key={doc.ref.id}
               className="Firestore-List-Item"
               tag={NavLink}
-              to={`${url}/${doc.ref.id}`}
+              to={`${url}/${encodeURIComponent(doc.ref.id)}`}
```
Comment (tstirrat) [inline src/components/Firestore/Collection.tsx:138]: Can you write a test for this? Thanks

File: src/components/Firestore/api.tsx [hunk 3]
```diff
@@ -58,7 +58,10 @@ export default class DatabaseApi extends RestApi implements FirestoreApi {
   private async getSubCollections(
     docRef: firestore.DocumentReference
   ): Promise<firestore.CollectionReference[]> {
-    const encodedPath = docRef.path; // TODO: Encode each segment.
+    const encodedPath = docRef.path
```
Comment (tstirrat) [inline src/components/Firestore/api.tsx:61]: A test for this would be helpful too

File: src/components/Firestore/Collection.tsx (hunk 1 above)
Comment (minho-comcom-ai) [reply src/components/Firestore/Collection.tsx:149]: @edupsousa I added a document with percent-encoded '/'(slash) included at Id. (e.g. 'hello%2fworld')
And it finally works without `decodeURIComponent()` in this line.
(@minhoryang)

**Comment by yuchenshi** [comment]
This is now out of date due to the `api.tsx` changes. If anyone still has bandwidth on this, feel free to update the PR or send us a new one. We just need to move the path encoding logic to the new files and add some tests.

**Comment by yuchenshi** [comment]
This is merged as https://github.com/firebase/firebase-tools-ui/pull/439. Thanks for the contribution!

**Pull request closed without merging**